packages/odyssey-react-mui/src/labs/GroupPicker.tsx

```diff
@@ -0,0 +1,247 @@
+/*!
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import type { AutocompleteGetTagProps } from "@mui/material/useAutocomplete";
+
+import {
+  Autocomplete as MuiAutocomplete,
+  AutocompleteProps as MuiAutocompleteProps,
+  Box,
+  Chip as MuiChip,
+  InputBase,
+  Avatar as MuiAvatar,
+} from "@mui/material";
+import { HTMLAttributes, memo, useCallback } from "react";
+
+import { AutocompleteProps } from "../Autocomplete";
+import { Field } from "../Field";
+import { Subordinate } from "../Typography";
+import { useOdysseyDesignTokens } from "../OdysseyDesignTokensContext";
+
+import { UserIcon, GridIcon, GroupIcon } from "../icons.generated";
+
+export type GroupPickerOptionType = {
+  id: string;
+  name: string;
+  description: string;
+  logo?: string;
+  usersCount?: number;
+  appsCount?: number;
+  groupPushMappingsCount?: number;
+};
+
+export type GroupPickerProps<
+  GroupPickerOptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined
+> = {
+  /**
+   * Hide selected options
+   */
+  filterSelectedOptions?: MuiAutocompleteProps<
+    GroupPickerOptionType,
+    HasMultipleChoices,
+    undefined,
+    IsCustomValueAllowed
+  >["filterSelectedOptions"];
+} & AutocompleteProps<
+  GroupPickerOptionType,
+  HasMultipleChoices,
+  IsCustomValueAllowed
+>;
+
+const GroupPicker = <
+  optionType extends GroupPickerOptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined
+>({
+  hasMultipleChoices,
+  isCustomValueAllowed,
+  isDisabled,
+  isLoading,
+  isOptional = false,
+  isReadOnly,
+  hint,
+  label,
+  onChange,
+  onInputChange,
+  options,
+  value,
+  filterSelectedOptions,
+  testId,
+}: GroupPickerProps<optionType, HasMultipleChoices, IsCustomValueAllowed>) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const isOptionEqualToValue = useCallback(
+    (sourceValue: optionType, targetValue: optionType) => {
+      return sourceValue.id === targetValue.id;
+    },
+    []
+  );
+
+  const getOptionLabel = useCallback((option: string | optionType) => {
+    return typeof option === "string" ? option : option.name;
+  }, []);
```

Comment:
Can we assign the `useCallback` a type of `UseAutocompleteProps["getOptionLabel"]`?